packages/marshal/src/rankOrder.js

```diff
@@ -44,9 +44,25 @@ const { entries, fromEntries, setPrototypeOf, is } = Object;
  */
 const sameValueZero = (x, y) => x === y || is(x, y);
 
+/**
+ * @deprecated use type-specific `compareNumerics` etc.
+ * @param {any} left
+ * @param {any} right
+ * @returns {RankComparison}
+ */
 export const trivialComparator = (left, right) =>
   // eslint-disable-next-line no-nested-ternary, @endo/restrict-comparison-operands
-  left < right ? -1 : left === right ? 0 : 1;
+  left < right ? -1 : left > right ? 1 : 0;
+
+/**
+ * @template {number | bigint} T
+ * @param {T} left
+ * @param {T} right
+ * @returns {RankComparison}
+ */
+export const compareNumerics = (left, right) =>
+  // eslint-disable-next-line no-nested-ternary, @endo/restrict-comparison-operands
+  left < right ? -1 : left > right ? 1 : 0;
```

Comment:
What about `NaN`?

Reply:
Hmm, good point. The current implementation of `trivialComparator` returns 1 for any comparison involving NaN, which is obviously broken but doesn't come up in either the Endo or agoric-sdk repositories. Still, we shouldn't leave that gap, so I've reverted the changes to `trivialComparator` and updated `compareNumerics` to support `compareRank`.